archive/tar_windows.go

```diff
@@ -1,6 +1,52 @@
 package archive
 
+import (
+	"archive/tar"
+	"os"
+	"strconv"
+	"syscall"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	symbolicLinkFlagAllowUnprivilegedCreate = 0x2
+
+	// MSWINDOWS pax vendor extensions
+	hdrMSWindowsPrefix = "MSWINDOWS."
+	hdrFileAttributes  = hdrMSWindowsPrefix + "fileattr"
+)
+
 func setUmask(newMask int) (oldMask int) {
 	// Not implemented on Windows
 	return 0
 }
+
+func createSymlink(hdr *tar.Header) error {
+	var flags uint32 = symbolicLinkFlagAllowUnprivilegedCreate
+	if attrStr, ok := hdr.PAXRecords[hdrFileAttributes]; ok {
+		attr, err := strconv.ParseUint(attrStr, 10, 32)
+		if err != nil {
+			return errors.Wrapf(err, "failed to parse file attributes for header %q", hdr.Name)
+		}
+		if attr&syscall.FILE_ATTRIBUTE_DIRECTORY != 0 {
+			flags |= syscall.SYMBOLIC_LINK_FLAG_DIRECTORY
+		}
+	}
+
+	name, err := syscall.UTF16PtrFromString(hdr.Name)
+	if err != nil {
+		return err
+	}
+	target, err := syscall.UTF16PtrFromString(hdr.Linkname)
+	if err != nil {
+		return err
+	}
+	return syscall.CreateSymbolicLink(name, target, flags)
+}
```

Comment:
It's a bummer to have to do all this to work around the divergent Windows/POSIX golang behavior. After reading that thread it sounds like a `wontfix` so `os.Symlink` won't change anytime soon. This implementation feels like it follows [one of the suggestions](https://github.com/golang/go/issues/39183#issuecomment-634251887) there:
> For the few programs that actually do need to create symlinks to nonexistent targets, windows.CreateSymbolicLink seems sufficient, and it's easy enough for users to write their own portable wrappers using +build constraints.